### DISCUSSION TITLE
Error in Building MOOSE

### DISCUSSION THREAD

**Discussion by sreihani** [idaholab/moose#28531] [Q&A Getting Started]
### Check these boxes if you have followed the posting rules.

- [X] I have consulted the [Posting Guidelines](https://github.com/idaholab/moose/discussions/18270).
- [X] I have searched the [Discussions Forum](https://github.com/idaholab/moose/discussions) and [MOOSE Framework Troubleshooting](https://mooseframework.inl.gov/help/troubleshooting.html) and have not found what I was looking for
- [X] Q&A Getting Started is the most appropriate category for my question (trouble installing, beginner user, ...)

### Issue or question about MOOSE

I am running into an error message, when trying to run the command:
make -j 6
Traceback (most recent call last):
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::dir_name(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::HITNodeView::data[abi:cxx11]() const'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::NodeView::data[abi:cxx11]() const'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::Value::operator[](std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::HITNodeView::to_string[abi:cxx11](bool*) const'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::strip_quotes(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::buildCompletionObject(wasp::DataObject&, std::ostream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, bool, bool, int)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::RPCStringToObject(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, wasp::DataObject&, std::ostream&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::HITNodeView::last_as_string[abi:cxx11](bool*) const'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::DataObject::operator[](std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::HITNodeView::first_child_by_name(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::buildLocationObject(wasp::DataObject&, std::ostream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, int, int, int)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::HITNodeView::path[abi:cxx11]() const'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::buildDocumentSymbolObject(wasp::DataObject&, std::ostream&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int, bool, int, int, int, int, int, int, int, int)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::objectToRPCString(wasp::DataObject&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::ostream&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::buildTextEditObject(wasp::DataObject&, std::ostream&, int, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::file_exists(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::lsp::buildDiagnosticObject(wasp::DataObject&, std::ostream&, int, int, int, int, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/apps/sreihani/projects/moose/framework/libmoose-opt.so: undefined reference to `wasp::HITNodeView::child_count_by_name(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, unsigned long) const'
collect2: error: ld returned 1 exit status
make: *** [/usr/apps/sreihani/projects/moose/test/moose_test-opt] Error 1

**Comment by GiudGiud** [comment]
Hello

This looks like your version of moose and wasp are not in sync.
Are you using the conda binaries for the dependencies?
If so the diagnostics script in moose/scripts should give you more information

Guillaume

**Comment by sreihani** [comment]
I am using GCC compiler and building the stack manually.

**Reply by milljm** [reply]
Can you print out the results of the diagnostic script? Sometimes it shows us some information that proves useful:
```bash
cd /usr/apps/sreihani/projects/moose/scripts
./diagnostics.sh
```

**Comment by sreihani** [comment]
I was able to find a solution for the issue and compiled the moose successfully. I think the order of modules loaded to the local environment is critical as there might be a conflict between some of them that can create Tcl command execution, However, I have added the output of diagnostic script below. If you have any recommendations, please let me know.

####################################################################################################
Influential Environment Variables

CC=gcc
CONDA_CHANNEL=https://conda.software.inl.gov/public
CURL_CA_BUNDLE=
F95=gfortran
LD_LIBRARY_PATH=/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib:/usr/local/gcc/alt/9.2.0/lib/gcc/x86_64-pc-linux-gnu/9.2.0:/usr/local/gcc/alt/9.2.0/lib64:/usr/local/gcc/alt/9.2.0/lib:/usr/local/src/git/2.19.0/lib:/usr/local/python/3.9.16/lib
MODULESHOME=/usr/share/Modules
MOOSE_JOBS=6
PATH=/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin:/usr/local/gcc/alt/9.2.0/bin:/usr/local/cmake/3.18.4/bin:/usr/local/src/git/2.19.0/bin:/usr/local/python/3.9.16/bin:/usr/local/dos2unix/7.4.0/bin:/usr/lib64/qt-3.3/bin:/usr/local/perlbrew/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin
REQUESTS_CA_BUNDLE=
SSL_CERT_FILE=
####################################################################################################
Compiler(s) (CC CXX FC F77 F90):

which $CC; /usr/local/gcc/alt/9.2.0/bin/gcc
$CC --version; gcc (GCC) 9.2.0
FAIL: gcc appears not to be an MPI wrapper

CXX     not set
FC      not set
F77     not set
F90     not set

FAIL: One or more compiler environment variables not set, or set incorrectly
####################################################################################################
Python Sanity Checks

/usr/bin/env python3 --version; (reporting as: Python 3.9.16) matches
which python3 python;

/usr/local/python/3.9.16/bin/python3 --version; == Python 3.9.16
Python 2.7.5
FAIL: Python 2.7.5
/usr/bin/python --version (reporting as )) != Python 3.9.16

This will likely result in the TestHarness failing. Or WASP/HIT parsing errors.
####################################################################################################
Python Modules (TestHarness, run-ability)


WARNING: missing module(s): jinja2

Either install the above packages, or perhaps you have yet
to activate the moose environment: conda activate moose

Missing 'warning' Python modules may cause miscellaneous runtime issues.

WARNING: One or more Python issues present.


checks FAILED

**Reply by milljm** [reply]
If the above report is based on the final set of `module load` commands completed... I would say MOOSE will ultimately fail to run properly.

Not sure if you've already corrected all the issues based on the diagnostic report, but they look simple enough:

```bash
export CC=mpicc CXX=mpicxx FC=mpif90 F90=mpif90 F77=mpif77
alias python=python3
./diagnostics.sh
```
_See note below about `alias python=python3`_

You can ignore the bit about `jinja2` not being available. You'll run into some minor things that might fail. But ultimately not needed.

Looks like my diagnostic script isn't properly understanding how to parse Python 2.7x. Its a bit difficult to read, but basically the error is thus:

Some scripts out in the wild which we depend on (like PETSc's configure scripts) may include shebangs that begin with `#!/usr/bin/env python` instead of the new proper way: `#!/usr/bin/env python3`. If this is still the case, your system may incur an issue due to the entire MOOSE dependency stack not using the same Python version throughout the build/run process.

Because this is an HPC system for which you probably don't have admin rights for, I _think_ the sollution which might work, is to create an alias for `python` to `python3`:

```bash
alias python=python3
```

This is only good for the terminal session you are running in (you'll need to perform this each and every time you start a terminal/log in). I am curious if after setting this, does `diagnostic.sh` pass.

**Reply by milljm** [reply]
Thanks for posting! I am constantly trying to improve the diagnostic script. It is meant to be a helper tool to figure out what is needed in order to fix the more common issues we answer on Discussions :)

The bit about:
```pre
CXX not set
FC not set
F77 not set
F90 not set

FAIL: One or more compiler environment variables not set, or set incorrectly
```
for example, I should be including what needs to be done

```pre
F90 not set

FAIL: One or more compiler environment variables not set, or set incorrectly

To solve this issue, you must set the above variables to point to your MPI wrapper:

export CC=mpicc CXX=mpicxx
```
etc etc

**Reply by sreihani** [reply]
Jason,

Here is the output of diagnostics test after what you suggested. It seems diagnostics.sh passes the compiler section but fails on python. However, python as an alias deploys python3. In my environment PYTHONPATH directs to 3.9.16. Can this help resolve the potential issue you mentioned? 

####################################################################################################
Influential Environment Variables

CC=mpicc
CONDA_CHANNEL=https://conda.software.inl.gov/public
CURL_CA_BUNDLE=
CXX=mpicxx
F77=mpif77
F90=mpif90
F95=gfortran
FC=mpif90
LD_LIBRARY_PATH=/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib:/usr/local/gcc/alt/9.2.0/lib/gcc/x86_64-pc-linux-gnu/9.2.0:/usr/local/gcc/alt/9.2.0/lib64:/usr/local/gcc/alt/9.2.0/lib:/usr/local/src/git/2.19.0/lib:/usr/local/python/3.9.16/lib
MODULESHOME=/usr/share/Modules
MOOSE_JOBS=6
PATH=/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin:/usr/local/gcc/alt/9.2.0/bin:/usr/local/cmake/3.18.4/bin:/usr/local/src/git/2.19.0/bin:/usr/local/python/3.9.16/bin:/usr/local/dos2unix/7.4.0/bin:/usr/lib64/qt-3.3/bin:/usr/local/perlbrew/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin
REQUESTS_CA_BUNDLE=
SSL_CERT_FILE=
####################################################################################################
Compiler(s) (CC CXX FC F77 F90):

which $CC; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpicc
$CC --version; gcc (GCC) 9.2.0
$CC -show; gcc -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi

which $CXX; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpicxx
$CXX --version; g++ (GCC) 9.2.0
$CXX -show; g++ -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi

which $FC; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpif90
$FC --version; GNU Fortran (GCC) 9.2.0
$FC -show; gfortran -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

which $F77; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpif77
$F77 --version; GNU Fortran (GCC) 9.2.0
$F77 -show; gfortran -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

which $F90; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpif90
$F90 --version; GNU Fortran (GCC) 9.2.0
$F90 -show; gfortran -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

OK
####################################################################################################
Python Sanity Checks

/usr/bin/env python3 --version; (reporting as: Python 3.9.16) matches
which python3 python;

/usr/local/python/3.9.16/bin/python3 --version; == Python 3.9.16
Python 2.7.5
FAIL: Python 2.7.5
/usr/bin/python --version (reporting as )) != Python 3.9.16

This will likely result in the TestHarness failing. Or WASP/HIT parsing errors.
####################################################################################################
Python Modules (TestHarness, run-ability)


WARNING: missing module(s): jinja2

Either install the above packages, or perhaps you have yet
to activate the moose environment: conda activate moose

Missing 'warning' Python modules may cause miscellaneous runtime issues.

WARNING: One or more Python issues present.


checks FAILED

**Reply by milljm** [reply]
I was wondering if aliasing would work. I am betting all the logic I have in the diagnostic script is ignoring the fact it should be looking at python3 instead.

I would ignore this for now, and continue to build the stack. If and when Python 2.7 becomes an issue, we will look into that.

**Reply by sreihani** [reply]
Jason,

Our admin created a link for python to make sure that potential issue would not occur. Now diagnostics.sh passes the test,

####################################################################################################
Influential Environment Variables

CC=mpicc
CONDA_CHANNEL=https://conda.software.inl.gov/public
CURL_CA_BUNDLE=
CXX=mpicxx
F77=mpif77
F90=mpif90
F95=gfortran
FC=mpif90
LD_LIBRARY_PATH=/usr/local/src/git/2.19.0/lib:/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib:/usr/local/gcc/alt/9.2.0/lib/gcc/x86_64-pc-linux-gnu/9.2.0:/usr/local/gcc/alt/9.2.0/lib64:/usr/local/gcc/alt/9.2.0/lib:/usr/local/python/3.9.16/lib
MODULESHOME=/usr/share/Modules
MOOSE_JOBS=6
PATH=/usr/local/src/git/2.19.0/bin:/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin:/usr/local/gcc/alt/9.2.0/bin:/usr/local/cmake/3.18.4/bin:/usr/local/python/3.9.16/bin:/usr/local/dos2unix/7.4.0/bin:/usr/lib64/qt-3.3/bin:/usr/local/perlbrew/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/sbin
REQUESTS_CA_BUNDLE=
SSL_CERT_FILE=
####################################################################################################
Compiler(s) (CC CXX FC F77 F90):

which $CC; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpicc
$CC --version; gcc (GCC) 9.2.0
$CC -show; gcc -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi

which $CXX; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpicxx
$CXX --version; g++ (GCC) 9.2.0
$CXX -show; g++ -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi

which $FC; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpif90
$FC --version; GNU Fortran (GCC) 9.2.0
$FC -show; gfortran -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

which $F77; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpif77
$F77 --version; GNU Fortran (GCC) 9.2.0
$F77 -show; gfortran -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

which $F90; /usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/bin/mpif90
$F90 --version; GNU Fortran (GCC) 9.2.0
$F90 -show; gfortran -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/include -pthread -I/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -L/usr/lib64 -Wl,-rpath -Wl,/usr/lib64 -Wl,-rpath -Wl,/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -Wl,--enable-new-dtags -L/usr/local/mpi/rh7/openmpi/4.1.1/gcc/9.2.0/lib -lmpi_usempif08 -lmpi_usempi_ignore_tkr -lmpi_mpifh -lmpi

OK
####################################################################################################
Python Sanity Checks

/usr/bin/env python3 --version; (reporting as: Python 3.9.16) matches
which python3 python;

/usr/local/python/3.9.16/bin/python3 --version; == Python 3.9.16
/usr/local/python/3.9.16/bin/python --version; == Python 3.9.16

OK
####################################################################################################
Python Modules (TestHarness, run-ability)


WARNING: missing module(s): jinja2

Either install the above packages, or perhaps you have yet
to activate the moose environment: conda activate moose

Missing 'warning' Python modules may cause miscellaneous runtime issues.

WARNING: One or more Python issues present.


checks PASSED

**Comment by sreihani** [comment]
I compiled the moose and did a ./run_tests -j  40. Here is the output:

Ran 4227 tests in 329.0 seconds. Average test time 1.6 seconds, maximum test time 124.9 seconds.
^[[1m^[[32m4215 passed^[[0m, ^[[1m281 skipped^[[0m, ^[[1m0 pending^[[0m, ^[[1m^[[31m12 FAILED^[[0m

There are 12 FAILED with the following errors:

misc/hit_cli.find/parameter_exists ^[[90m...........................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.find/parameter_does_not_exist ^[[90m...................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.find/positive_pattern ^[[90m...........................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.find/negative_pattern ^[[90m...........................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.find/additional_pattern ^[[90m.........................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.find/additional_pattern_negative ^[[90m................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.merge ^[[90m...........................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.diff/vanilla ^[[90m....................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.diff/color ^[[90m......................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.diff/verbose ^[[90m....................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.diff/verbose_color ^[[90m..............................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
misc/hit_cli.diff/common ^[[90m.....................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m

Could you let me know if the FAILED cases would be detrimental to the whole processes in moose?

**Reply by GiudGiud** [reply]
these are all tied to hit, the parser
so it s safe to say it wont affect results

**Reply by sreihani** [reply]
Thanks. 
I am wondering if there is any other tool to open exit files of moose other than VSCode recommended on the web?

**Reply by GiudGiud** [reply]
input files can be opened with any text editor
output files, it depends on the type. We use mostly paraview for exodus and excel or paraview for CSVs

**Reply by sreihani** [reply]
Sure, I know that. I was referring to this page: https://mooseframework.inl.gov/help/development/VSCode.html
In the videos of MOOSE workshops, VSCode is recommended for visualizing the output binary files.

**Reply by GiudGiud** [reply]
we dont use binary for output, expect for some versions of checkpoints

**Reply by sreihani** [reply]
For a tutorial test, moose generated two files (step1_out.e & step1_in.e). vi editor shows both have binary content. Am I correct? Also, does output files have a specific format & name convention?

**Reply by GiudGiud** [reply]
these are exodus files, you can open them with paraview

**Reply by milljm** [reply]
I imagine so, but I can't tell for certain. If you would scroll back a ways in the terminal that produces this error, you will see details for each error.

You can also run just one of these tests:
```bash
./run_tests --re=misc/hit_cli.find/parameter_does_not_exist
```
And copy/paste the error. There are two parts to every erring test. This page describes it best: https://mooseframework.inl.gov/help/troubleshooting.html#failingtests

**Reply by sreihani** [reply]
Here is the output for the command:
```pre
./run_tests --re=misc/hit_cli.find/parameter_does_not_exist

^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0mWorking Directory: /usr/apps/SoTeRiA/sreihani/projects/moose/test/tests/misc/hit_cli
^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0mRunning command: ./hit_wrapper.py gold/parameter_does_not_exist find -v Kernels/\*/variable input1.i input2.i
^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0mFailed to locate hit executable.
^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0m
^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0m################################################################################
^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0mTester failed, reason: CODE 1
^[[31mmisc/hit_cli.find/parameter_does_not_exist: ^[[0m
misc/hit_cli.find/parameter_does_not_exist ^[[90m...................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m


Final Test Results:
--------------------------------------------------------------------------------------------------------------
misc/hit_cli.find/parameter_does_not_exist ^[[90m...................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
--------------------------------------------------------------------------------------------------------------
Ran 1 tests in 4.1 seconds. Average test time 0.0 seconds, maximum test time 0.0 seconds.
^[[1m0 passed^[[0m, ^[[1m0 skipped^[[0m, ^[[1m0 pending^[[0m, ^[[1m^[[31m1 FAILED^[[0m
```

**Reply by milljm** [reply]
Thats weird to me... I would imagine "hit" missing would result in a great many more tests failing... @GiudGiud  any ideas?

**Reply by sreihani** [reply]
Here are results for always_ok & always_bad:
```
tests/test_harness.always_ok ^[[90m..................................................................^[[0m ^[[36m[OVERSIZED]^[[0m ^[[32mOK^[[0m
--------------------------------------------------------------------------------------------------------------
Ran 1 tests in 1.7 seconds. Average test time 1.4 seconds, maximum test time 1.4 seconds.
^[[1m^[[32m1 passed^[[0m, ^[[1m0 skipped^[[0m, ^[[1m0 pending^[[0m, ^[[1m0 failed^[[0m

always_bad:

tests/test_harness.always_ok ^[[90m..............................................................................^[[0m ^[[32mOK^[[0m
^[[31mtests/test_harness.always_bad: ^[[0mWorking Directory: /usr/apps/SoTeRiA/sreihani/projects/moose/test/tests/test_harness
^[[31mtests/test_harness.always_bad: ^[[0mRunning command: false
^[[31mtests/test_harness.always_bad: ^[[0m
^[[31mtests/test_harness.always_bad: ^[[0m################################################################################
^[[31mtests/test_harness.always_bad: ^[[0mTester failed, reason: CODE 1
^[[31mtests/test_harness.always_bad: ^[[0m
tests/test_harness.always_bad ^[[90m................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m


Final Test Results:
--------------------------------------------------------------------------------------------------------------
tests/test_harness.always_ok ^[[90m..............................................................................^[[0m ^[[32mOK^[[0m
tests/test_harness.always_bad ^[[90m................................................................^[[0m ^[[31mFAILED (CODE 1)^[[0m
--------------------------------------------------------------------------------------------------------------
Ran 2 tests in 0.6 seconds. Average test time 0.1 seconds, maximum test time 0.3 seconds.
^[[1m^[[32m1 passed^[[0m, ^[[1m0 skipped^[[0m, ^[[1m0 pending^[[0m, ^[[1m^[[31m1 FAILED^[[0m
```

**Reply by GiudGiud** [reply]
`hit` is probably just for python parsing of the inputs. It's not the C++ parser. So most inputs should be fine.
I would navigate to the hit folder in contrib and try to build there, or just ignore it

**Reply by milljm** [reply]
the `always_ok always_bad` worked as expected. Your issue is different.

The important bit:
```pre
Failed to locate hit executable.
```

Or ignore it, as @GiudGiud mentioned. As it seems to only cause a very small subset of tests to fail.

**Reply by sreihani** [reply]
Here is output to (Just FYI, I am installing MOOSE from source)

./run_tests -j 6 --re=misc/hit_cli.find/parameter_does_not_exist

misc/hit_cli.find/parameter_does_not_exist: Working Directory: /usr/apps/SoTeRiA/sreihani/projects/moose/test/tests/misc/hit_cli
misc/hit_cli.find/parameter_does_not_exist: Running command: ./hit_wrapper.py gold/parameter_does_not_exist find -v Kernels/\*/variable input1.i input2.i
misc/hit_cli.find/parameter_does_not_exist: Failed to locate hit executable.
misc/hit_cli.find/parameter_does_not_exist:
misc/hit_cli.find/parameter_does_not_exist: ################################################################################
misc/hit_cli.find/parameter_does_not_exist: Tester failed, reason: CODE 1
misc/hit_cli.find/parameter_does_not_exist:
misc/hit_cli.find/parameter_does_not_exist ................................................... FAILED (CODE 1)


Final Test Results:
--------------------------------------------------------------------------------------------------------------
misc/hit_cli.find/parameter_does_not_exist ................................................... FAILED (CODE 1)
--------------------------------------------------------------------------------------------------------------
Ran 1 tests in 15.3 seconds. Average test time 0.1 seconds, maximum test time 0.1 seconds.
0 passed, 0 skipped, 0 pending, 1 FAILED
